webapp/store/survey/hooks/index.js

```diff
@@ -32,3 +36,58 @@ export const useNodeDefsByUuids = (uuids) => Survey.getNodeDefsByUuids(uuids)(us
 
 // ==== Categories
 export const useCategoryByUuid = (uuid) => Survey.getCategoryByUuid(uuid)(useSurvey())
+
+const useSurveyItem = ({ type }) => {
+  const surveyId = useSurveyId()
+  const [state, setState] = useState([])
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const { data } = await axios.get(`/api/survey/${surveyId}/${type}`, {
+        params: { draft: true, validate: false },
+      })
+      setState(data.list)
+    }
+    fetchData()
+  }, [surveyId])
+
+  const getItemsById = useCallback(
+    () => (state || []).reduce((byId, item) => ({ ...byId, [ObjectUtils.getUuid(item)]: { ...item } }), {}),
+    [state]
+  )
+
+  const getItemsByUuid = useCallback(
```

Comment:
this should be renamed into getItemByUuid right?

Reply:
yes!!! hehe thanks!